run.c

```diff
@@ -186,18 +186,20 @@ void run(void)
         if(WriteSnapshot || WriteFOF) {
             int snapnum = All.SnapshotFileCount++;
 
-            /* the accel may have created garbage -- collect them before checkpointing! */
-            force_tree_free();
-            slots_gc();
-            force_tree_rebuild();
+            /* The accel may have created garbage -- collect them before checkpointing!
+             * Tree will be auto-rebuilt if gc collected particles,
+             * but we should rebuild the active list.*/
+            int compact[6] = {0};
+            if(slots_gc(compact))
```

Comment:
`slots_gc(need_compact, &base_moved);`?

Reply:
What does the base_moved argument do?

Reply:
as a named return value ..

Reply:
But it sounds like rebuild_activelist can be an eventhandler registered by the timebin manager module. Then you get get rid of the call to rebuild_activelist here.

Reply:
I can do this instead of the event handler, but returning the number of items moved is a fairly standard C library convention, so I quite like it like this.